From: "seven.steps@gmail.com" <seven.steps@gmail.com>
Date: Thu, 24 Nov 2011 21:40:40 +0100
Subject: Fix format string use

Origin: vendor
Bug-Debian: http://bugs.debian.org/565064
Forwarded: no
Last-Update: 2011-11-24
---
 score.C |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/score.C b/score.C
index 928324a..9e3d4b0 100644
--- a/score.C
+++ b/score.C
@@ -337,7 +337,7 @@ Score::DisplayScore(int kind_of_game)
     }
     // Always read the latest high score
     readfile();
-    sprintf(format, res_strings[0]);
+    sprintf(format, "%s", res_strings[0]);
     prev_time = -1;
     for(int i=0; i<SCORENUM; i++) {
         this_time = rec[i].hour * 3600 + rec[i].min * 60 + rec[i].sec;
